flower/utils/results/stores/database.py

```diff
@@ -1,13 +1,107 @@
+import json
+import pickle
 from collections.abc import Iterator
+from typing import Any
 
-from celery.backends.database import DatabaseBackend
+import kombu.serialization
+import sqlalchemy as sa
+from celery.backends.database import DatabaseBackend, TaskExtended
+from sqlalchemy.orm.session import Session
 
-from flower.utils.results.result import ResultIdWithResultPair
+from flower.utils.results.result import ResultIdWithResultPair, Result
 from flower.utils.results.stores import AbstractBackendResultsStore
 
 
 class DatabaseBackendResultsStore(AbstractBackendResultsStore[DatabaseBackend]):
+    """
+    Results store capable of reading from Celery's supported `DatabaseBackend`, which uses SQLAlchemy models to persist
+    tasks in a SQL database.
+    """
     def results_by_timestamp(self, limit: int | None = None, reverse: bool = True) -> Iterator[
         ResultIdWithResultPair
     ]:
-        raise NotImplementedError()
+        query_limit = self.max_tasks_in_memory
+        if limit is not None and limit < query_limit:
+            query_limit = limit
+
+        session: Session
+        with self.backend.ResultSession() as session:
+            ordering = TaskExtended.date_done.desc() if reverse else TaskExtended.date_done.asc()
+            task_select_query = sa.select(TaskExtended).order_by(ordering).limit(query_limit)
+            for task in session.execute(task_select_query).scalars():
+                result = self._map_task_to_result(task)
+                yield result.task_id, result
+
+    def _map_task_to_result(self, task: TaskExtended) -> Result:
+        """
+        Convert a `TaskExtended` ORM object into our shared `Result` data structure. This class assumes the usage of
+        `TaskExtended` in order to query the "taskmeta" table, since `TaskExtended` queries can successfully return
+        full data for both tasks that were saved with `result_extended=True` and those that were saved with
+        `result_extended=False`.
+
+        Because we want to support both extended and non-extended tasks, we need a way to figure out whether the
+        provided task was _actually_ extended or not at the time it was saved. We can do this by looking at the `name`
+        field. When a task is saved under `result_extended=True`, then it will have a name referencing the name of the
+        function. Otherwise, that field will be null and we know that it was `result_extended=False`.
```

Comment:
Clever... how stable do you think this check is?

Reply:
in general i'd like to advocate against choosing to have the semi-predicate problem, if we can include a bit for is_result_extended, i would much prefer using it, especially when we're in python and have tuples/dicts given to us for free.

Reply:
I think it's pretty stable. I've dug deep into the Celery implementation, and it seems like when you have extended=True, you get `name`, `args`, and `kwargs`. I figured that `name` is the best of those to use, since it may be possible for `args` or `kwargs` to be null for some reason.

Reply:
> to be null for some reason

I assume the obvious would be something like:

```python
@task
def my_function():
    pass

my_function.delay()
```

Reply:
Yes, although I'm not sure if it Celery would store NULLs for a no-arg task, or if it would store `args=()` and `kwargs={}`. My big fear was that this might vary in different scenarios, so I figured `name` was a safer choice.